mindsdb/integrations/handlers/confluence_handler/confluence_table.py

```diff
@@ -16,119 +16,98 @@ class ConfluencePagesTable(APITable):
                 if a_where[0] != "=":
                     raise ValueError("Unsupported where operation for space")
-                pages_kwargs["space"] = space_name
-            elif a_where[1] not in ["id", "space"]:
-                raise ValueError(f"Unsupported where argument {a_where[1]}")
-
-        confluence_pages_records = self.handler.connect().get_all_pages_from_space(
-            space_name, start=0, limit=total_results, expand="body.storage"
-        )
-        confluence_pages_df = pd.json_normalize(confluence_pages_records)
-
-        for a_where in conditions:
-            if a_where[1] == "id":
-                id = a_where[2]
+            elif a_where[1] == "id":
+                page_id = a_where[2]
                 if a_where[0] != "=":
                     raise ValueError("Unsupported where operation for id")
-                confluence_pages_df = confluence_pages_df[
-                    confluence_pages_df.id == id
-                ]
-                
+            elif a_where[1] not in ["id", "space"]:
+                raise ValueError(f"Unsupported where argument {a_where[1]}")
 
-        def extract_space(input_string):
-            parts = input_string.split('/')
-            return parts[-1]
+        if space_name is None:
+            raise ValueError("Space name must be provided in the WHERE clause")
 
-        confluence_pages_df["space"] = confluence_pages_df["_expandable.space"].apply(
-            extract_space
-        )
+        if page_id is not None:
+            try:
+                page = self.handler.connect().get_page_by_id(page_id, expand='body.storage,space')
+                confluence_pages_records = [page] if page['space']['key'] == space_name else []
+            except Exception as e:
+                logger.error(f"Error fetching page with ID {page_id}: {str(e)}")
+                confluence_pages_records = []
+        else:
+            confluence_pages_records = self.handler.connect().get_all_pages_from_space(
+                space_name, start=0, limit=total_results, expand="body.storage,space"
+            )
 
-        confluence_pages_df.columns = confluence_pages_df.columns.str.replace(
-            "body.storage.value", "body"
-        )
+        if not confluence_pages_records:
+            return pd.DataFrame(columns=self.get_columns())
 
-        confluence_pages_df = confluence_pages_df[self.get_columns()]
+        confluence_pages_df = pd.json_normalize(confluence_pages_records)
 
-        if "space" in pages_kwargs:
-            confluence_pages_df = confluence_pages_df[
-                confluence_pages_df.space == pages_kwargs["space"]
-            ]
+        
+        confluence_pages_df.columns = confluence_pages_df.columns.str.replace("body.storage.value", "body")
+        confluence_pages_df['space'] = confluence_pages_df['space.key']
+        
+        available_columns = set(confluence_pages_df.columns).intersection(set(self.get_columns()))
```

Comment:
Can you explain the reason for doing this? It is not clear.

Reply:
I  added  it in case
The API response might not contain all columns defined in get_columns  or it 
 might contain additional columns not defined in get_columns
should i remove it?

Reply:
That does not make sense. We should know what columns (attributes) are returned by the API response.

Reply:
@MinuraPunchihewa  i have made the required changes please take look